middleman-core/lib/middleman-core/util/data.rb

```diff
@@ -118,9 +118,17 @@ def build_regex(frontmatter_delims)
       # @return [Hash]
       Contract String, Pathname => Hash
       def parse_yaml(content, full_path)
+        permitted_classes = [Date, Symbol]
         c = begin
           ::Middleman::Util.instrument 'parse.yaml' do
-            ::YAML.load(content)
+            allowed_parameters = ::YAML.method(:safe_load).parameters
+            if allowed_parameters.include? [:key, :permitted_classes]
+              ::YAML.safe_load(content, permitted_classes: permitted_classes)
+            elsif allowed_parameters.include? [:key, :whitelist_classes]
+              ::YAML.safe_load(content, whitelist_classes: permitted_classes)
+            else
+              ::YAML.safe_load(content, permitted_classes)
+            end
```

Comment:
How many ruby versions would we need to drop support for to replace this with the current version (`::YAML.safe_load(content, permitted_classes: permitted_classes)`)?

Is it just ruby 2.5? Or 2.6 too?

Given that [2.6 and 2.5 are end of life](https://www.ruby-lang.org/en/downloads/branches/) I wonder if the middleman maintainers would consider dropping support for them so we could avoid this compatibility workaround?

Reply:
Sorry, we won't be dropping any Ruby support. The vast majority of middleman apps are ancient and I prefer to keep those projects running smoothly over new Ruby 3.0 features which really don't come into play with a static site generator.

Reply:
No worries, the workaround looks fine to me - worth a little bit of complexity if there are still folks on older ruby.